### PR TITLE
core[minor]: Add status field to tool

### PR DESCRIPTION
**Pull request by bracesproul**
Added new `status` field to tools.

TODO in separate PR:
- [update anthropic to use this field](https://github.com/langchain-ai/langchain/pull/24628/files#diff-e2dea7070f0cc79e3fc72e08f7a809d2ae17ae8924794bd917c72d37dc65cc1c)
- doc section on `status` field.
- update LangGraph `ToolNode` to use `status` field.

### CODE REVIEW

File: langchain-core/src/tools/index.ts [hunk 1]
```diff
@@ -212,7 +212,8 @@ export abstract class StructuredTool<
     let result;
     try {
       result = await this._call(parsed, runManager, config);
-    } catch (e) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (jacoblee93) [inline langchain-core/src/tools/index.ts:215]: Changes here seem fine but unrelated?

**Comment by jacoblee93** [comment]
Looks good!

Only thing I'll flag is that streamed chunks will always have `success` set to true while tool messages may not

**Comment by bracesproul** [comment]
> Looks good!
> 
> Only thing I'll flag is that streamed chunks will always have `success` set to true while tool messages may not

Going to push a refactor to leave undefined when concatenating unless it's passed.

File: langchain-core/src/tools/index.ts (hunk 1 above)
Comment (bracesproul) [reply langchain-core/src/tools/index.ts:215]: forgot to remove when I deleted the error handling code from this PR.

**Pull request merged**